src/utils/fetch.utils.ts

```diff
@@ -1,6 +1,54 @@
 import * as utils from '@/utils'
+import { type ITokenPriceData } from '@/interfaces'
 
-export const fetchTokenPrice = async (tokenAddress: string, chainId: string, host?: string) => {
+type IMobulaMarketData = {
+    id: number
+    market_cap: number
+    market_cap_diluted: number
+    liquidity: number
+    price: number
+    off_chain_volume: number
+    volume: number
+    volume_change_24h: number
+    volume_7d: number
+    is_listed: boolean
+    price_change_24h: number
+    price_change_1h: number
+    price_change_7d: number
+    price_change_1m: number
+    price_change_1y: number
+    ath: number
+    atl: number
+    name: string
+    symbol: string
+    logo: string
+    rank: number
+    contracts: {
+        address: string
+        blockchain: string
+        blockchainId: string
+        decimals: number
+    }[]
+    total_supply: string
+    circulating_supply: string
+    decimals?: number
+    priceNative: number
+    native: {
+        name: string
+        address: string
+        decimals: number
+        symbol: string
+        type: string
+        logo: string
+        id: number
+    }
+}
+
```

Comment:
QUestion: Do we need such  a detailed type? What if they change their response? In practice we only care about a few of these fields

Reply:
we don't need it to be so detailed, I just added the current one they return and if they change it this wil lbe outdated